x/orderbook/amount.go

```diff
@@ -13,3 +18,35 @@ func NewAmountp(whole int64, fractional int64) *Amount {
 		Fractional: fractional,
 	}
 }
+
+func (a *Amount) Clone() *Amount {
+	return &Amount{
+		Whole:      a.Whole,
+		Fractional: a.Fractional,
+	}
+}
+
+// Validate "borrowed" from coin.Coin
+func (a *Amount) Validate() error {
+	if a == nil {
+		return errors.Wrap(errors.ErrEmpty, "amount")
+	}
+	if a.Whole < coin.MinInt || a.Whole > coin.MaxInt {
+		return errors.Wrap(errors.ErrOverflow, "whole")
```

Comment:
Why not [errors.ErrCurrency](https://github.com/iov-one/weave/blob/7b38cf4be9d3bea44b76454f4b941120c2d00f22/errors/errors.go#L66)? What is the use cases of ```errors.ErrCurrency```?

Reply:
ErrCurrency is for invali currency codes (not `[A-Z]{3,4}`) or adding mis-matched currencies

Overflow is an explicit call for integer overflow, or exceeding acceptable bounds that we impose to avoid overflows (this was copied from coin.Coin.Validate btw)